source/ShipInfoPanel.cpp

```diff
@@ -169,9 +169,41 @@ bool ShipInfoPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command,
 	else if(panelState.CanEdit() && key == 'D')
 	{
 		if(shipIt->get() != player.Flagship())
-			GetUI()->Push(new Dialog(this, &ShipInfoPanel::Disown, "Are you sure you want to disown \""
+		{
+			map<const Outfit*, int> uniqueOutfits;
+			for(const auto &it : shipIt->get()->Outfits())
+				if(it.first->Attributes().Get("unique") > 0)
+					uniqueOutfits[it.first] = it.second;
+			for(const auto &it : shipIt->get()->Cargo().Outfits())
+				if(it.first->Attributes().Get("unique") > 0)
+					uniqueOutfits[it.first] += it.second;
```

Comment:
```suggestion
			auto AddToUniques = [&] (const std::map<const Outfit *, int> &outfits)
			{
				for(const auto &it : outfits)
					if(it.first->Attributes().Get("unique") > 0)
						uniqueOutfits[it.first] += it.second;
			};
			AddToUniques(shipIt->get()->Outfits());
			AddToUniques(shipIt->get()->Cargo().Outfits());
```
to avoid repeating the same code twice